src/main/java/org/opensearch/flowframework/model/Config.java

```diff
@@ -59,12 +78,17 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
     public static Config parse(XContentParser parser) throws IOException {
         String masterKey = null;
         Instant createTime = Instant.now();
+        String tenantId = null;
 
         ensureExpectedToken(XContentParser.Token.START_OBJECT, parser.currentToken(), parser);
         while (parser.nextToken() != XContentParser.Token.END_OBJECT) {
             String fieldName = parser.currentName();
             parser.nextToken();
+            logger.info("Parsing field: {}, token: {}, value: {}", fieldName, parser.currentToken(), parser.text());
             switch (fieldName) {
+                case TENANT_ID_FIELD:
+                    tenantId = parser.text();
```

Comment:
This could be `null`.  Use `parser.objectText()` here which will either be `null` or a `String` you can either cast or `toString()` the object to.